Sources/UtiliKit/ActiveLabel/ActiveLabel.swift

```diff
@@ -0,0 +1,285 @@
+    /// Trailing Inset for the last activity line. Default is 0.
+    @IBInspectable public var finalLineTrailingInset: CGFloat = 0
+    /// Line Length in points for the last activity line. If finalLineTralingInset is set to greater than 0 this value is not used. Default is 0.
+    @IBInspectable public var finalLineLength: CGFloat = 0
+    /// This color is the darkest area of the line seen during activity animation. Default is (233,231,237) Gray.
+    @IBInspectable public var loadingViewColor: UIColor = UIColor(red: 233.0/255.0, green: 231.0/255.0, blue: 237.0/255.0, alpha: 1.0)
+    /// The height of each activity line. Default is 8.
+    @IBInspectable public var loadingLineHeight: CGFloat = 8
+    /// Vertical spacing between each activity line when 2 or more lines are displayed. Default is 14.
+    @IBInspectable public var loadingLineVerticalSpacing: CGFloat = 14
+    /// The duration of the gradient animation applied to the activity lines. Default is 2.4.
+    @IBInspectable public var loadingAnimationDuration: Double = 2.4
+    /// The delay that is applied before each animation begins. Default is 0.4.
+    @IBInspectable public var loadingAnimationDelay: Double = 0.4
+    /// Shows the loading views in the Storyboard by default since text of a label can't be nil in a Storybaord. Default is true.
+    @IBInspectable private var showLoadingViewsInStoryboard: Bool = true
+    
+    private static let locationKeyPath = "locations"
+    
+    private var isDisplayingInStoryboard: Bool = false
+    private(set) var isLoading: Bool = false {
+        didSet {
+            guard oldValue != isLoading else { return }
+            
+            if isLoading {
+                text = nil
+                if !isHidden {
+                    showLoadingViews()
+                }
+            } else {
+                loadingViews.forEach({ $0.removeFromSuperview() })
+                loadingViews.removeAll()
+            }
+        }
+    }
+    private var loadingViews: [UIView] = []
+    private var maskOffset: CGFloat {
+        // The offset is enough to make sure that the gradient doesn't originally show on screen.
+        return bounds.width * 0.4
+    }
+    
+    /**
+     When text is set to nil the loading views will display. When text is non-nill the loading views will be hidden.
+     */
+    override var text: String? {
+        didSet {
+            textDidUpdate()
+        }
+    }
+    
+    override var isHidden: Bool {
+        didSet {
+            if isHidden {
+                hideLoadingViews()
+            } else if !isHidden && isLoading {
+                showLoadingViews()
+            }
+        }
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        defer {
+            textDidUpdate()
+        }
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        defer {
```

Comment:
Do we need `defer` (since there's no other logic inside these initializers)?

Reply:
I don't think we do. I'll pull it and test to make sure.